jax/_src/random.py

```diff
@@ -1327,10 +1327,13 @@ def categorical(key: KeyArray,
     shape = tuple(shape)
     _check_shape("categorical", shape, batch_shape)
 
-  sample_shape = shape[:len(shape)-len(batch_shape)]
+  shape_prefix = shape[:len(shape)-len(batch_shape)]
+  logits_shape = list(shape[len(shape) - len(batch_shape):])
+  logits_shape.insert(axis % len(logits_arr.shape), logits_arr.shape[axis])
+  logits_arr= jnp.broadcast_to(logits_arr, logits_shape)
```

Comment:
insert space before =

Reply:
Done! Also I deleted that line just now.